build-android/app/src/main/java/com/google/firebase/codelab/friendlychat/SignInActivity.kt

```diff
@@ -0,0 +1,109 @@
+ * limitations under the License.
+ */
+package com.google.firebase.codelab.friendlychat
+
+import android.content.Intent
+import android.os.Bundle
+import android.util.Log
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import com.google.android.gms.auth.api.signin.GoogleSignIn
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount
+import com.google.android.gms.auth.api.signin.GoogleSignInClient
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions
+import com.google.android.gms.common.api.ApiException
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.auth.GoogleAuthProvider
+import com.google.firebase.auth.ktx.auth
+import com.google.firebase.codelab.friendlychat.databinding.ActivitySignInBinding
+import com.google.firebase.ktx.Firebase
+
+class SignInActivity : AppCompatActivity() {
+    private lateinit var mBinding: ActivitySignInBinding
+    private lateinit var mSignInClient: GoogleSignInClient
+
+    // Firebase instance variables
+    private lateinit var mFirebaseAuth: FirebaseAuth
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        // This codelab uses View Binding
+        // See: https://developer.android.com/topic/libraries/view-binding
+        mBinding = ActivitySignInBinding.inflate(layoutInflater)
+        setContentView(mBinding.root)
+
+        // Set click listeners
+        mBinding.signInButton.setOnClickListener { signIn() }
+
+        // Configure Google Sign In
+        val gso = GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+            .requestIdToken(getString(R.string.default_web_client_id))
+            .requestEmail()
+            .build()
+        mSignInClient = GoogleSignIn.getClient(this, gso)
+
+        // Initialize FirebaseAuth
+        mFirebaseAuth = Firebase.auth
+    }
+
+    private fun signIn() {
+        val signInIntent = mSignInClient.signInIntent
+        startActivityForResult(signInIntent, RC_SIGN_IN)
+    }
+
+    public override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+
+        // Result returned from launching the Intent in signIn()
+        if (requestCode == RC_SIGN_IN) {
+            val task = GoogleSignIn.getSignedInAccountFromIntent(data)
+            try {
+                // Google Sign In was successful, authenticate with Firebase
+                val account = task.getResult(ApiException::class.java)
+                firebaseAuthWithGoogle(account)
+            } catch (e: ApiException) {
+                // Google Sign In failed, update UI appropriately
+                Log.w(TAG, "Google sign in failed", e)
+            }
+        }
+    }
+
+    private fun firebaseAuthWithGoogle(acct: GoogleSignInAccount?) {
+        Log.d(TAG, "firebaseAuthWithGoogle:" + acct!!.id)
```

Comment:
Adding the `!!` here negates the use of the optional `GoogleSignInAccout?` since the app will now crash if it's null.

So we could either:
  * Make the parameter required non-null
  * Use optional `?.id` syntax to access the id